.infra/rdev/values.yaml

```diff
@@ -22,4 +22,4 @@ stack:
         - name: DATA_LOCATOR_DOMAIN
           value: $(__ARGUS_STACK_INGRESS_HOST)
         - name: CXG_BUCKET_PATH
-          value: hosted-cellxgene-dev
+          value: hosted-cellxgene-staging
```

Comment:
@hspitzley-czi I'm trying to test a fix by changing Explorer BE from hitting dev to staging, but I'm getting:

> Unable to discover s3 region name from s3://hosted-cellxgene-staging (type=value_error)

In https://argo.prod.platform.si.czi.technology/applications/certain-bull?resource=&node=%2FPod%2Fargus-single-cell-explorer-rdev%2Fcertain-bull-stack-explorer-7654d9cc58-8nzrg%2F0&tab=logs

Is that because rdev role permission doesn't have access to staging bucket?

The bug I'm trying to address is that rdev gene info endpoint always returns `null` 🤔 

CURL

```
curl 'https://certain-bull.dev-sc.dev.czi.team/cellxgene/s3_uri/s3%253A%252F%252Fhosted-cellxgene-dev%252Fsuper-cool-spatial.cxg/api/v0.3/geneinfo?geneID=&gene=AR' \
  -H 'accept: application/json' \
  -H 'accept-language: en-US,en;q=0.9,zh-TW;q=0.8,zh;q=0.7,zh-CN;q=0.6' \
  -H 'cookie: argus_sticky_session=1734566048.418.1662.83615|e3f9a91a7a0db7bca0ae9c8322a67838' \
  -H 'dnt: 1' \
  -H 'priority: u=1, i' \
  -H 'referer: https://certain-bull.dev-sc.dev.czi.team/d/super-cool-spatial.cxg/' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

whereas on staging it returns actual info:

```
curl 'https://api.cellxgene.staging.single-cell.czi.technology/cellxgene/s3_uri/s3%253A%252F%252Fhosted-cellxgene-staging%252Fpbmc3k.cxg/api/v0.3/geneinfo?geneID=&gene=ACD' \
  -H 'accept: application/json' \
  -H 'accept-language: en-US,en;q=0.9,zh-TW;q=0.8,zh;q=0.7,zh-CN;q=0.6' \
  -H 'cookie: argus_sticky_session=1734565349.449.38.739565|df4dba4aefd2407066eb62a1488673d2' \
  -H 'dnt: 1' \
  -H 'origin: https://cellxgene.staging.single-cell.czi.technology' \
  -H 'priority: u=1, i' \
  -H 'referer: https://cellxgene.staging.single-cell.czi.technology/' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-site' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

Reply:
Hmm, that seems likely. I'm not sure what the application code is running to produce that error but looking at the role used by rdev I'm not seeing permission for the staging bucket
https://github.com/chanzuckerberg/single-cell-explorer/blob/main/.infra/common.yaml#L24

Reply:
I see! I think the code that produces the error is this line in the Explorer BE code:

https://github.com/chanzuckerberg/single-cell-explorer/blob/31fd26bf0e1f73d3616d5bbbd720aef1712551be/server/common/config/config_model.py#L201

My hypothesis is that the Explorer BE is returning `null` for gene info query now, because behind the scene Explorer GE actually queries Data Portal's dev gene info endpoint for data, and _I believe_ dev data portal endpoint is no longer working, thus us seeing the issue

https://cellxgene.dev.single-cell.czi.technology/ <-- gets 503 service not available now

So I'm thinking the solution here is for Explorer rdev to switch from hitting dev to staging and thus needing permission to staging bucket

Should I just change the common `iam` from `dev` to `staging` ?

`arn:aws:iam::699936264352:role/data_portal_staging_explorer`

Let me try that out! Thanks for pointing me to the code!

Reply:
You might want to check if there are other differences in the permissions between the dev role and staging role. You could probably pretty easily add permission to the dev role for the staging bucket

Reply:
Ah thanks for the callout! Okay let me test if using staging role actually works, if so I'll just add staging bucket permission to dev role 🙆‍♂️

Reply:
Huh interesting - changing the iam role to staging still produces the same error 🤔 

What else am I missing 😆

Reply:
Hmm, do you know which of these `None`s is returned? If it's from the boto client call it might be helpful to log the actual error

Reply:
Ah yeah good point! Let me add some logs thank you!

Reply:
Ok just added logs will share what I find! Thanks!

Reply:
Hmm yeah seems like it's just getting 403 Forbidden

> Attempting to discover region for bucket: hosted-cellxgene-staging
ClientError occurred while accessing bucket hosted-cellxgene-staging: An error occurred (403) when calling the HeadBucket operation: Forbidden
1 validation error for AppConfigModel

🤔

Reply:
printing iam role now to be super sure I'm indeed assuming staging role 😆

Reply:
Ah interesting, the assume role is this:

`arn:aws:iam::699936264352:user/dev-explorer-jheath-argus-demo-deleteme`

>Attempting to discover region for bucket: hosted-cellxgene-staging
Assumed role details: {'UserId': 'AIDA2F53T2CQLEU7J4CM4', 'Account': '699936264352', 'Arn': 'arn:aws:iam::699936264352:user/dev-explorer-jheath-argus-demo-deleteme', 'ResponseMetadata': {'RequestId': '15405df9-d935-4464-8c92-8229ebe10bad', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': '15405df9-d935-4464-8c92-8229ebe10bad', 'content-type': 'text/xml', 'content-length': '436', 'date': 'Thu, 19 Dec 2024 23:05:19 GMT'}, 'RetryAttempts': 0}}

Reply:
@hspitzley-czi it seems like the assumed role is `dev-explorer-jheath-argus-demo-deleteme` instead of the specified `data_portal_staging_explorer` 😆 Is it being temporarily overwritten somewhere?

Thank you!

Reply:
I can't find anything that overwrites it or any reference to the `dev-explorer-jheath-argus-demo-deleteme` role, continuing to investigate...

Reply:
Huh yeah github org wide search didn't return anything o_o

https://github.com/search?q=org:chanzuckerberg%20jheath-argus-demo-deleteme&type=code

Reply:
Ooh it looks like the access key and secret key set with argus secrets belongs to that user. (pass `-r` to see the values)
```
» argus list secrets --app single-cell-explorer --env rdev                                                                                                                                                                                          hspitzley@MacBookPro
Listing environment variables on environment rdev
+-----------------------+----------+-------------+
|          KEY          |  VALUE   |   SOURCE    |
+-----------------------+----------+-------------+
| AWS_ACCESS_KEY_ID     | ******** | environment |
| AWS_SECRET_ACCESS_KEY | ******** | environment |
+-----------------------+----------+-------------+
```

Reply:
There was some reason we needed it to assume a user instead of a role (I'm forgetting why right now) but this gives us a path forward. You could copy those values elsewhere so you can restore them later and then overwrite them with the corresponding values from staging and I think that will make it able to access the staging bucket

Reply:
oh fantastic!! Thanks so much for helping me out with this 🤩 🙏 That's really good to know 💡 

Okay will do and retest 🫡 

Thanks again!